### DISCUSSION TITLE
afterAll() is not firing

### DISCUSSION THREAD

**Discussion by jeytii** [pestphp/pest#502] [Q&A]
All of my test files under `Feature` folder have to refresh the database upon completion and I want to just declare it on `Pest.php` instead to apply the DRY rule. The problem is... it's skipping the `afterAll()` method.

```
// tests/Pest.php

uses()
    ->beforeAll(function() {
        dump('beforeall');
    })
    ->beforeEach(function() {
        dump('beforeeach');
    })
    ->afterEach(function() {
        dump('aftereach');
    })
    ->afterAll(function() {
        dump('afterall'); // Doesn't appear on the output
    })
    ->in('Feature');
```

Am I missing something?

**Comment by fabio-ivona** [comment]
Hi @jeytii , I just tried on a project of mine:

![image](https://user-images.githubusercontent.com/8792274/158958085-3139d564-2de1-424e-8f21-bccfc092f13a.png)

`afterAll` seems to work, it gets printed in the last dump

could you setup a simple public project that reproduces your issue?

**Reply by jeytii** [reply]
It's working now. I think it was because I installed `pestphp/pest-plugin-laravel` without installing `pestphp/pest` first. :man_facepalming: 

Thanks.

**Comment by raz-iacob** [comment]
I just tried afterAll and is not executing for me. I have pest version 3.8.2

// tests/Pest.php

```
pest()->extend(TestCase::class)
    ->use(RefreshDatabase::class)
    ->beforeEach(function (): void {
        Http::preventStrayRequests();
        $this->freezeTime();
    })
    ->afterAll(function (): void {
        dump('here');
    })
    ->in('Feature', 'Unit');
  ```
  
   I don't see any dump in my console. If I move the dump function to beforeEach, then it works. What am I missing here?
   
   I tried this in 2 different projects.
   
   Also is there a way to execute cleanup code at the end of my tests? after all the test files have executed? From what I understand, afterAll hook will run after each file.
